TwoWayBinding/TwoWayBinding/EditProfileViewController.swift

```diff
@@ -0,0 +1,207 @@
+
+    let datePicker = UIDatePicker()
+    let genderPicker = UIPickerView()
+
+    private let disposeBag = DisposeBag()
+    private let dismissTapGesture = UITapGestureRecognizer()
+
+    init(profile: Profile) {
+        viewModel = EditProfileViewModel(profile: profile)
+        super.init(nibName: nil, bundle: nil)
+    }
+
+    override convenience init(nibName nibNameOrNil: String?, bundle nibBundleOrNil: Bundle?) {
+        self.init(profile: Profile())
+    }
+
+    required init?(coder aDecoder: NSCoder) {
+        fatalError("init(coder:) has not been implemented")
+    }
+
+    override func viewDidLoad() {
+        super.viewDidLoad()
+
+        dismissTapGesture.addTarget(self, action: #selector(EditProfileViewController.endEditing))
+        view.addGestureRecognizer(dismissTapGesture)
+
+        usernameTextField.rx.text <-> viewModel.username >>> disposeBag
+        emailTextField.rx.text <-> viewModel.email >>> disposeBag
+        firstNameTextField.rx.text <-> viewModel.firstName >>> disposeBag
+        lastNameTextField.rx.text <-> viewModel.lastName >>> disposeBag
+
+        usernameTextField.rx.backgroundColor <- viewModel.usernameBackgroundColor >>> disposeBag
+        emailTextField.rx.backgroundColor <- viewModel.emailBackgroundColor >>> disposeBag
+
+        configureDateOfBirthField()
+        configureGenderField()
+
+        logButton.rx.tap.subscribe { _ in
+            print(self.viewModel)
+        } >>> disposeBag
+    }
+
+    private func configureDateOfBirthField() {
+        datePicker.datePickerMode = .date
+        dateOfBirthTextField.inputView = datePicker
+        datePicker.rx.date <-> viewModel.dateOfBirth >>> disposeBag
+        dateOfBirthTextField.rx.text <- viewModel.dateOfBirthString >>> disposeBag
+    }
+
+    private func configureGenderField() {
+        // TODO: Create RxDatasource for this.
+        // TODO: Extend the delegate proxy to include titleForRow:forComponent:
+        genderPicker.dataSource = self
+        genderPicker.delegate = self
+        genderPicker.rx.itemSelected.map { (row, component) -> Gender in
+            let gender = Gender(rawValue: row)
+            return gender ?? .other
+            }.bindTo(viewModel.gender) >>> disposeBag
+
+        genderTextField.inputView = genderPicker
+        genderTextField.rx.text <- viewModel.genderString >>> disposeBag
+    }
+
+    dynamic private func endEditing() {
+        view.endEditing(true)
+    }
+
+    // MARK: UIPickerViewDataSource
+
+    func pickerView(_ pickerView: UIPickerView, numberOfRowsInComponent component: Int) -> Int {
+        return 3
+    }
+
+    func numberOfComponents(in pickerView: UIPickerView) -> Int {
+        return 1
+    }
+
+    // MARK: UIPickerViewDelegate
+
+    func pickerView(_ pickerView: UIPickerView, titleForRow row: Int, forComponent component: Int) -> String? {
+        let gender =  Gender(rawValue: row) ?? .other
+        return gender.displayText
+    }
+}
+
+import RxCocoa
```

Comment:
Agreed on most of the files.  I wanted to show how easy it would be to create a Reactive extension.  By moving it to a different file, it feels like it obscures that the code was created locally.  How about moving this to the top of the file and commenting why it is here?